### PR TITLE
Add --feed-lock-timeout option

### PR DESCRIPTION
**Pull request by timopollmeier**
**What**:
This new option sets a number of seconds to retry for if the feed is
locked in contexts (like migration or rebuilds) that do not retry on
their own (like automatic syncs).

**Why**:
This is intended to help automated upgrades of GVM modules where
ospd-openvas needs to be restarted, which locks the feed while the
VTs are reloaded.

**How did you test it**:
By doing the following:
* locking the feed lockfile and running `gvmd --rebuild-scap` - this should fail immediately.
* retrying with `gvmd --rebuild-scap --feed-lock-timeout 15` with the lock still active - this should generate a gvmd log message "Feed is currently locked..." and fail about 15 seconds later.
* running `gvmd --rebuild-scap --feed-lock-timeout 15` with the lock still active and releasing the lock while gvmd is waiting - this should leave a "Feed is currently locked..." gvmd log message and start the rebuild once the lock is released. 
* running `gvmd --rebuild-scap --feed-lock-timeout 15` without locking - this should start the rebuild immediately without a "Feed is currently locked..." log message.

**Checklist**:

- [ ] Tests
- [x] [CHANGELOG](https://github.com/greenbone/gvmd/blob/master/CHANGELOG.md) Entry

### CODE REVIEW

File: doc/gvmd.8 [hunk 1]
```diff
@@ -182,9 +185,6 @@ Time out tasks that are more than TIME minutes overdue. -1 to disable, 0 for min
 \fB--secinfo-commit-size=\fINUMBER\fB\f1
 During CERT and SCAP sync, commit updates to the database every NUMBER items, 0 for unlimited.
 .TP
-\fB--slave-commit-size=\fINUMBER\fB\f1
-During slave updates, commit after every NUMBER updated results and hosts, 0 for unlimited.
-.TP
```
Comment (bjoernricks) [inline doc/gvmd.8:187]: Has this been removed by accident or on purpose? Maybe it belongs to another PR?

File: src/manage.c [hunk 2]
```diff
@@ -6345,7 +6419,7 @@ gvm_migrate_secinfo (int feed_type)
       return -1;
     }
 
-  ret = feed_lockfile_lock (&lockfile);
```
Comment (bjoernricks) [inline src/manage.c:6348]: What about the old function? Is is still used or can it be removed?

File: src/manage.c [hunk 3]
```diff
@@ -5995,6 +6025,50 @@ feed_lockfile_lock (lockfile_t *lockfile)
...
+                         " will retry until %s.",
+                         __func__, iso_time (&timeout_end));
+            }
+          gvm_sleep (1);
+        }
+      else if (ret)
+        {
+          return ret;
+        }
+    } while (ret);
```
Comment (bjoernricks) [inline src/manage.c:6064]: What's the meaning of ret here? 1 == locked? If yes maybe you could rename it to somthing like is_locked.

File: src/gvmd.c [hunk 4]
```diff
@@ -1833,6 +1834,12 @@ gvmd (int argc, char** argv)
           &feed_lock_path,
           "Sets the path to the feed lock file.",
           "<path>" },
+        { "feed-lock-timeout", '\0', 0, G_OPTION_ARG_INT,
+          &feed_lock_timeout,
+          "Sets the number of seconds to retry for if the feed is locked"
+          " in contexts (like migration or rebuilds) that do not retry"
+          " on their own (like automatic syncs). Defaults to 0.",
```
Comment (bjoernricks) [inline src/gvmd.c:1841]: I would explain the meaning of 0 too (i guess it is nowait).

**Review by bjoernricks — CHANGES_REQUESTED** [review]

File: doc/gvmd.8 (hunk 1 above)
Comment (timopollmeier) [reply doc/gvmd.8:187]: This option is supposed to be removed. I think it was added back in by accident in a backport or it was just forgotten before.

File: src/manage.c (hunk 2 above)
Comment (timopollmeier) [reply src/manage.c:6348]: The old function is still used in the automatic sync that has essentially its own retry loop.
I've also changed `feed_lockfile_lock_timeout` to use it now so writing the timestamp doesn't have to be duplicated.

File: src/manage.c (hunk 3 above)
Comment (timopollmeier) [reply src/manage.c]: I've renamed the variable and added a few comments about the conditions.

File: src/gvmd.c (hunk 4 above)
Comment (timopollmeier) [reply src/gvmd.c]: I've updated the text to mention that the default 0 means there is no retry.

**Review by bjoernricks — APPROVED** [review]

**Comment by timopollmeier** [comment]
@mergifyio refresh

**Comment by timopollmeier** [comment]
@mergifyio backport gvmd-20.08
